Sources/ArgumentParser/Utilities/Platform.swift

```diff
@@ -144,17 +190,38 @@ extension Platform {
 #else
     let err = ioctl(STDOUT_FILENO, TIOCGWINSZ, &w)
 #endif
-    let width = Int(w.ws_col)
-    let height = Int(w.ws_row)
-    guard err == 0 else { return defaultTerminalSize }
-    return (width: width > 0 ? width : defaultTerminalSize.width,
-            height: height > 0 ? height : defaultTerminalSize.height)
+    guard err == 0 else { return (width: nil, height: nil) }
+
+    let width = Int(w.ws_col), height = Int(w.ws_row)
+    
+    return (width: width > 0 ? width : nil,
+            height: height > 0 ? height : nil)
 #endif
   }
   
+  /// Returns the current terminal size, or the default if the size is unavailable.
+  static func terminalSize() -> (width: Int, height: Int) {
+    let specifiedSize = self.userSpecifiedTerminalSize()
+    
+    // Avoid needlessly calling ioctl() if a complete override is in effect
+    if let specifiedWidth = specifiedSize.width, let specifiedHeight = specifiedSize.height {
+        return (width: specifiedWidth, height: specifiedHeight)
+    }
+    
+    // Get the size self-reported by the terminal, if available
+    let reportedSize = self.reportedTerminalSize()
+    
+    // As it isn't required that both width and height always be specified
+    // together, either by the user or the terminal itself, they are
+    // handled separately.
+    return (
+      width: specifiedSize.width ?? reportedSize.width ?? defaultTerminalSize.width,
+      height: specifiedSize.height ?? reportedSize.height ?? defaultTerminalSize.height
+    )
+  }
+  
   /// The current terminal size, or the default if the width is unavailable.
   static var terminalWidth: Int {
-    terminalSize().width
+    self.terminalSize().width
```

Comment:
I appreciate the explicit `self`

Reply:
Suffice to say I wish there was an `.enableUpcomingFeature()` flag called `"DisallowImplicitSelf"` 🤣

Reply:
+10000000